### ISSUE TITLE
arguments-differ when overriding a regular method using class/static methods

### ISSUE THREAD

**Issue by noirbee** [PyCQA/pylint#6946]
### Bug description

Since 2.13, pylint raises a warning when overriding a regular bound method as a class / static method in children classes:

```python
class BaseWithSelf:
    def my_method(self, arg: str) -> None:
        print(self, arg)


# Overriding a regular method as a classmethod should be valid
class ClassMethodChild(BaseWithSelf):
    @classmethod
    def my_method(cls, arg: str) -> None:
        print(cls, arg)


# Overriding a regular method as a staticmethod should be valid
class StaticMethodChild(BaseWithSelf):
    @staticmethod
    def my_method(arg: str) -> None:
        print(arg)
```

However, replacing a regular bound method with a class/static method in children classes does not modify the API: you'd still be expected to call it as a bound method (e.g. `StaticMethodChild().my_method("hello")`).

This was introduced in PR https://github.com/PyCQA/pylint/pull/5412 / commit 856c9ef13f08e35a40162fe0a4760b8dce372302

I tried to fiddle with the code without much success, not sure how this could be handled.

### Command used

```shell
pylint test_staticmethod_redef.py
```

### Pylint output

```shell
************* Module test_staticmethod_redef
test_staticmethod_redef.py:9:4: W0221: Number of parameters was 2 in 'BaseWithSelf.my_method' and is now 2 in overridden 'ClassMethodChild.my_method' method (arguments-differ)
test_staticmethod_redef.py:16:4: W0221: Number of parameters was 2 in 'BaseWithSelf.my_method' and is now 1 in overridden 'StaticMethodChild.my_method' method (arguments-differ)

-----------------------------------
Your code has been rated at 7.78/10
```

### Expected behavior

No warning is raised.

### Pylint version

```shell
pylint 2.15.0-dev0
astroid 2.11.6
Python 3.10.5 (main, Jun  8 2022, 09:26:22) [GCC 11.3.0]

AFAICT it affects all pylint versions since 2.13
```

**Comment by jacobtylerwalls** [comment]
Hi @noirbee thanks for the report. IIUC, this is the same issue as discussed in #6865. @cdce8p felt strongly this was the correct behavior. Let's continue the discussion there if you like.
